util/src/main/scala/geotrellis/util/LocalBytesStreamer.scala

```diff
@@ -0,0 +1,60 @@
+package geotrellis.util
+
+import java.io._
+import java.nio.ByteBuffer
+import java.nio.channels.FileChannel
+import java.nio.channels.FileChannel.MapMode._
+
+/**
+ * This class extends [[BytesStreamer]] by reading chunks from a given local path. This
+ * allows for reading in of files larger than 4gb into GeoTrellis.
+ *
+ * @param path: A String that is the path to the local file.
+ * @param chunkSize: An Int that specifies how many bytes should be read in at a time.
+ * @return A new instance of LocalBytesStreamer
+ */
+class LocalBytesStreamer(path: String, val chunkSize: Int) extends BytesStreamer {
+  private val f: File = new File(path)
+  private val inputStream: FileInputStream = new FileInputStream(f)
+  private val channel: FileChannel =  inputStream.getChannel
+
+  def objectLength: Long = channel.size
+
+  def getArray(start: Long, length: Long): Array[Byte] = {
```

Comment:
This is just a modified version of [slurp](https://github.com/geotrellis/geotrellis/blob/master/util/src/main/scala/geotrellis/util/Filesystem.scala#L52). Unfortunately, I wasn't able to find another way of doing ranged, byte reads from a local file stream.